### ISSUE TITLE
Class-named services cannot be used for hooks

### ISSUE THREAD

**Issue by aschempp** [contao/core-bundle#1176]
Due to the `class_exists` check in [`System::importStatic`](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/System.php#L207) (and `System::import`) it is not possible to register a hook listener service if the class name is used as service name. Do we really need that `class_exists` check?

**Comment by Toflar** [comment]
Nope, `container->has()` is enough. +1 for dropping that (as a bugfix!).

**Comment by leofeyer** [comment]
We have added the check to prevent services from accidentally overwriting existing classes, which would then no longer be loaded. So no, `$container->has()` is not enough.

**Comment by discordier** [comment]
@leofeyer This won't work for native Symfony 3.3 applications anymore as the suggested service name IS the class name.
See [here](https://symfony.com/blog/new-in-symfony-3-3-optional-class-for-named-services) and [here](https://symfony.com/doc/current/service_container/3.3-di-changes.html)

So in the end you will have to alias an symfony service to something else to make it usable in Contao then?
I'm all up for using only `$container->has()`.

**Comment by aschempp** [comment]
I think the only and major issue we had initially was with our `Session` class and Symfony's session. Because `System::import('Session')` would take the service instead of our class. So we gotta make sure about that …

**Comment by aschempp** [comment]
I think the best approach is to check for a backslash in the name.

```php
if ($container->has($name) && (false !== strpos($name, '\') || !class_exist($name)) {
}
```

**Comment by leofeyer** [comment]
Fixed in 9361ea95f748044faa2f05e36322c5ff8e488111.
